### PR TITLE
Add test for jsdoc in navigation bar

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

File: tests/cases/fourslash/navigationBarJsDoc.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/// <reference path="fourslash.ts"/>
+
+// @Filename: foo.js
+/////** @typedef {(number|string)} NumberLike */
+/////** @typedef {(string|number)} */
+////const x = 0;
+
+verify.navigationBar([
+  {
+    "text": "NumberLike",
```
Comment (mhegazy) [inline tests/cases/fourslash/navigationBarJsDoc.ts]: should not there be a `<global>` node?

Comment (andy-ms) [inline tests/cases/fourslash/navigationBarJsDoc.ts]: The current implementation is different for javascript, so that will be something fixed by #8958.

Comment (mhegazy) [inline tests/cases/fourslash/navigationBarJsDoc.ts]: Thanks!

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**